core/src/test/java/org/apache/calcite/rex/RexExecutorTest.java

```diff
@@ -378,4 +378,28 @@ public void run() {
   interface Action {
     void check(RexBuilder rexBuilder, RexExecutorImpl executor);
   }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5949">[CALCITE-5949]
+   * RexExecutable should properly handle invalid constant expressions in reduction</a>.
```

Comment:
@arkanovicz could you please align this comment with the latest title / commit message ("RexExecutable should return unchanged original expressions when it fails")?
Then please squash commits into a single one, and I'll proceed with the merge

Reply:
@arkanovicz  kind reminder about this minor (old) remark regarding the ticket title on this javadoc.

Reply:
@arkanovicz we are approaching the preparation for the next release, could you please carry out these last details ⬆️  in order to merge this PR? thanks